### PR TITLE
63 create employer model

### PR DESCRIPTION
**Pull request by bartorl**

### CODE REVIEW

File: controllers/authController.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const Applicant = require('../models/applicantModel');
-const sessionController = require('./sessionController')
+const Employer = require('../models/empolyerModel');
```
Comment (macibol775) [inline controllers/authController.js:2]: empolyer?

File: controllers/authController.js [hunk 2]
```diff
@@ -22,12 +23,37 @@ const register_put = async(req, res)=>{
...
         }
-            // console.log(e);
-            res.json({ errors });
-        }
+        res.json({ errors });
+    }
         
 };
 
+//shpuld be move to admin controller when created
```
Comment (macibol775) [inline controllers/authController.js:31]: I know that it is a comment, but please...

File: controllers/authController.js [hunk 3]
```diff
@@ -22,12 +23,37 @@ const register_put = async(req, res)=>{
...
+const createEmployer_put = async(req, res) =>{
+    const { email, password, name, surname, permissionLevel } = req.body;
+
+    try{
+        const employer = await Employer.create({ email, password, name, surname,permissionLevel });
+        console.log("New employer %s created", email, permissionLevel);
```
Comment (macibol775) [inline controllers/authController.js:37]: does it work properly?

File: controllers/authController.js [hunk 4]
```diff
@@ -22,12 +23,37 @@ const register_put = async(req, res)=>{
...
+        res.status(201).jason({ redirect: 'login '});
```
Comment (macibol775) [inline controllers/authController.js:38]: JASON?!?

Comment (macibol775) [reply controllers/authController.js:38]: path without space

File: controllers/employerController.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const Employer = require('../models/empolyerModel');
```
Comment (macibol775) [inline controllers/employerController.js:1]: empolyer?

File: controllers/employerController.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+const panel_get = (req, res) => {
+    if(req.session.authenticated){
+        res.render('employer/employerPanel', { title: 'employerPanel' });
```
Comment (macibol775) [inline controllers/employerController.js:5]: we've decided that frontend should be in polish, so why title is in english?

File: controllers/sessionController.js [hunk 7]
```diff
@@ -6,12 +6,17 @@ const saveApplicantInfoToSession = (req, name, surname, email, _id) => {
...
+const saveEmployerInfoToSession = (req, name, surname, email, permLevel, _id) => {
+    req.session.employer = {  name: name, surname: surname, _id: _id, email: email, level: permLevel };
+};
+
 const destroySession = (req) => {
     req.session.destroy();
 };
 
 module.exports = {
   authenticationAfterloggingIn,
```
Comment (macibol775) [inline controllers/sessionController.js:18]: Change L in Login to capital letter (here and above)

File: models/empolyerModel.js
Comment (macibol775) [inline models/empolyerModel.js:1]: RENAME!

File: models/empolyerModel.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        throw Error('Incorrect password');
+    } else {
+        throw Error('Invalid email');
+    }
+};
+
+const Employer = mongoose.model('Employer', employerSchema);
+
+module.exports = Employer;
```
Comment (macibol775) [inline models/empolyerModel.js:69]: (endl)

**Review by macibol775 — CHANGES_REQUESTED** [review]
generally typos

File: controllers/authController.js (hunk 3 above)
Comment (bartorl) [reply controllers/authController.js:37]: Yes

**Review by macibol775 — APPROVED** [review]

File: controllers/authController.js [hunk 9]
```diff
@@ -22,12 +23,37 @@ const register_put = async(req, res)=>{
                 }
             });
         }
-            // console.log(e);
-            res.json({ errors });
-        }
+        res.json({ errors });
```
Comment (MasterCoookie) [inline controllers/authController.js:26]: send status

File: controllers/authController.js [hunk 10]
```diff
@@ -22,12 +23,37 @@ const register_put = async(req, res)=>{
...
+         }
+        
+        if(e.errors) {
+            Object.values(e.errors).forEach(({ properties }) => {
+                if (properties.message) {
+                    errors.push(properties.message);
+                }
+            });
+        }
+        res.json({ errors });
```
Comment (MasterCoookie) [inline controllers/authController.js:52]: send code

File: controllers/authController.js [hunk 11]
```diff
@@ -37,17 +63,32 @@ const login_post = async (req, res) => {
...
-                console.log("Logged in");
-                sessionController.authenticationAfterloggingIn(req);
-                sessionController.saveApplicantInfoToSession(req, applicant.name, applicant.surname, applicant.email, applicant._id);
-                res.status(202).json({ redirect: 'profile' });
-            } else {
+            const isEmployer = await Employer.findOne({ email: email });
+            if(isEmployer) {
+                const employer = await Employer.login(email, password);
+                if(employer) {
+                    console.log("Logged in as employer");
```
Comment (MasterCoookie) [inline controllers/authController.js:70]: In those kind of msg, say who logged in

File: controllers/employerController.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+const Employer = require('../models/employerModel');
+
+const panel_get = (req, res) => {
+    if(req.session.authenticated){
+        res.render('employer/employerPanel', { title: 'panelPracownika' });
```
Comment (MasterCoookie) [inline controllers/employerController.js:5]: brak spacji i duża pierwsza litera

File: controllers/sessionController.js [hunk 13]
```diff
@@ -1,17 +1,22 @@
...
+const authenticationAfterLoggingIn = (req) => {
     req.session.authenticated = true;
 };
 
 const saveApplicantInfoToSession = (req, name, surname, email, _id) => {
     req.session.applicant = { name: name, surname: surname, _id: _id, email: email };    
 };
 
+const saveEmployerInfoToSession = (req, name, surname, email, permLevel, _id) => {
+    req.session.employer = {  name: name, surname: surname, _id: _id, email: email, level: permLevel };
```
Comment (MasterCoookie) [inline controllers/sessionController.js:10]: In those kind of assignment you can use shortened version. If key name matches value name u can write
`{ name, surname, _id } itd...`

File: models/employerModel.js [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+		validate: [pwdValid, 'Your password must contain both letters (lowercase and uppercase) and numbers']
+	},
+    permissionLevel: {
+        type: Number,
+        default: 1
+    },
+});
+
+employerSchema.pre('save', async function(next) {  
+    if (!this.isModified('password')){
```
Comment (MasterCoookie) [inline models/employerModel.js:46]: Duplicated code, but will pass this time

**Review by MasterCoookie — CHANGES_REQUESTED** [review]
Send statuses and fix typo in title

File: controllers/sessionController.js (hunk 13 above)
Comment (bartorl) [reply controllers/sessionController.js:10]: Will use it next time if possible

**Review by MasterCoookie — APPROVED** [review]

**Pull request merged**